src/components/pager/pager-navigation.component.js

```diff
@@ -1,138 +1,53 @@
 import NumberInput from '../../__experimental__/components/number';
 import Events from '../../utils/helpers/events';
-import baseTheme from '../../style/themes/base';
-import { isClassic } from '../../utils/helpers/style-helper';
-
-const PagerNavigation = (props) => {
-  const {
-    theme, setCurrentThemeName, setCurrentPage, onPagination
-  } = props;
-
-  setCurrentThemeName(theme);
-
+import PagerNavigationLink from './pager-navigation-link.component';
+
+const PagerNavigation = ({
+  currentPage,
+  pageSize,
+  totalRecords,
+  onFirst,
+  onPrevious,
+  onLast,
+  onNext,
+  setCurrentPage,
+  onPagination,
+  ...props
+}) => {
   const updatePageFromInput = (ev) => {
     let newPage = Math.abs(Number(ev.target.value));
 
     if (Number(newPage) === 0 || Number.isNaN(newPage)) newPage = '1';
-    else if (newPage > maxPages()) newPage = String(maxPages());
+    else if (newPage > maxPages()) newPage = maxPages();
 
-    onPagination(String(newPage), props.pageSize, 'input');
-    setCurrentPage(String(newPage));
+    setCurrentPage(Number(newPage));
   };
 
   function maxPages() {
-    const totalRecordsCount = props.totalRecords >= 0 ? props.totalRecords : 0;
-
-    if (props.pageSize && props.pageSize !== '0' && totalRecordsCount > 0) {
-      return Math.ceil(totalRecordsCount / props.pageSize);
-    }
-    return 1;
-  }
-
-  function navArrowChange(step) {
-    const newPage = String(Number(props.currentPage) + step);
-    const desc = step === 1 ? 'next' : 'previous';
-    onPagination(newPage, props.pageSize, desc);
-  }
-
-  function navArrow(step) {
-    const arrowProps = {
-      'data-element': 'next-page',
-      type: 'dropdown'
-    };
-
-    function disabled() {
-      if (maxPages() === 1) {
-        return true;
-      }
-      if (step === -1) {
-        return props.currentPage === '1';
-      }
-      return props.currentPage === String(maxPages());
-    }
-
-    return (
-      <PagerButtonWrapperStyles disabled={ disabled() } next={ step === 1 }>
-        <Icon
-          onClick={ () => {
-            if (!disabled()) { navArrowChange(step); }
-          } }
-          { ...arrowProps }
-        />
-      </PagerButtonWrapperStyles>
-    );
-  }
-
-  function navLink(type) {
-    const currentPage = Number(props.currentPage);
-    const navLinkConfig = {
-      first: {
-        text: I18n.t('pager.first', { defaultValue: 'First' }),
-        destination: '1'
-      },
-      last: {
-        text: I18n.t('pager.last', { defaultValue: 'Last' }),
-        destination: String(maxPages())
-      },
-      next: {
-        text: I18n.t('pager.next', { defaultValue: 'Next' }),
-        destination: String(currentPage + 1)
-      },
-      back: {
-        text: I18n.t('pager.previous', { defaultValue: 'Previous' }),
-        destination: String(currentPage - 1)
-      }
-    };
-
-    const { destination, text } = navLinkConfig[type];
+    const totalRecordsCount = totalRecords >= 0 ? totalRecords : 0;
```

Comment:
we don't need this line, you could just check in the conditional below if `totalRecords` is more than 0